ir/fail_test.go

```diff
@@ -0,0 +1,39 @@
+package ir
```

Comment:
Currently, we don't really test fail situations in other parts of the `ir` package. I think it would make sense to add these kind of test cases, either to cover most error cases, or skip them for most.

For now, I'd simply remove this test case unless there is clear motivation why it should exist in the absence of tests for other failure situations. @pwaller do you have any thoughts on the matter?